src/citrine/builders/auto_configure.py

```diff
@@ -0,0 +1,558 @@
+    def table(self) -> Optional[GemTable]:
+        """Get the default GEM table, if it was created by this object."""
+        return self._table
+
+    @property
+    def predictor(self) -> Optional[Predictor]:
+        """Get the default predictor, if it was created by this object."""
+        return self._predictor
+
+    @property
+    def predictor_evaluation_workflow(self) -> Optional[PredictorEvaluationWorkflow]:
+        """Get the predictor evaluation workflow, if it was created by this object."""
+        return self._predictor_evaluation_workflow
+
+    @property
+    def predictor_evaluation_execution(self) -> Optional[PredictorEvaluationExecution]:
+        """Get the predictor evaluation execution, if it was created by this object."""
+        return self._predictor_evaluation_execution
+
+    @property
+    def design_space(self) -> Optional[DesignSpace]:
+        """Get the design space used during the auto configuration steps."""
+        return self._design_space
+
+    @property
+    def design_workflow(self) -> Optional[DesignWorkflow]:
+        """Get the default design workflow, if it was created by this object."""
+        return self._desing_workflow
+
+    @property
+    def design_execution(self) -> Optional[DesignExecution]:
+        """Get the design execution, if it was created by this object."""
+        return self._design_execution
+
+    @property
+    def assets(self):
+        """Get all assets created by this object."""
+        initial_assets = [
+            self.table_config, self.table, self.predictor,
+            self.predictor_evaluation_workflow, self.predictor_evaluation_execution,
+            self.design_space, self.design_workflow, self.design_execution
+        ]
+        return [asset for asset in initial_assets if asset is not None]
+
+    def execute(self, *, score: Score) -> DesignExecution:
+        """[ALPHA] Execute a design execution given a provided score.
+
+        A design workflow must be present after auto-configuration to execute the score.
+        This method can be called after modifying auto-configured assets
+        to create a new, custom design execution.
+
+        Parameters
+        ----------
+        score: Score
+            Scoring function used to rank candidates during design execution.
+            Must contain objectives/constraints with matching descriptor keys
+            to those appearing in predictor responses.
+
+        """
+        if self.design_workflow is None:
+            raise ValueError("Design workflow is missing, cannot execute score.")
+        execution = self.design_workflow.design_executions.trigger(score)
+        self._design_execution = execution
+        return self.design_execution
+
+    def from_material(
+            self,
+            *,
+            material: Union[str, UUID, LinkByUID, MaterialRun],
+            evaluator: Optional[PredictorEvaluator] = None,
+            design_space: Optional[DataSourceDesignSpace] = None,
```

Comment:
```suggestion
            design_space: Optional[DesignSpace] = None,
```
Why limit to a `DataSourceDesignSpace`?  A user might want to re-use an existing product design space.

This applies to all the `DataSourceDesignSpace` signatures.

Reply:
This has been updated for any design space type.